pydatastructs/trees/_backend/cpp/BinaryTreeTraversal.hpp

```diff
@@ -171,6 +171,47 @@ static PyObject* BinaryTreeTraversal__out_order(BinaryTreeTraversal* self, PyObj
     return visit;
 }
 
+static PyObject* BinaryTreeTraversal_morris_in_order_traversal(BinaryTreeTraversal* self, PyObject *args) {
+    PyObject* node = PyObject_GetItem(args, PyZero);
+    if (node == Py_None) {
+        node = self->tree->root_idx;
+    }
+
+    PyObject* traversal = PyList_New(0);
+    ArrayForTrees* tree = self->tree->tree;
+    long current = PyLong_AsLong(node);
+
+    while (current != Py_None) {
+        TreeNode* current_node = reinterpret_cast<TreeNode*>(tree->_one_dimensional_array->_data[current]);
+        if (current_node->left == Py_None) {
+            // If there's no left child, visit the current node
+            PyList_Append(traversal, reinterpret_cast<PyObject*>(current_node));
+            current = PyLong_AsLong(current_node->right);
+        } else {
+            // Find the in-order predecessor (rightmost node in the left subtree)
+            long predecessor = PyLong_AsLong(current_node->left);
+            TreeNode* predecessor_node = reinterpret_cast<TreeNode*>(tree->_one_dimensional_array->_data[predecessor]);
+            while (predecessor_node->right != Py_None && predecessor_node->right != node) {
+                predecessor = PyLong_AsLong(predecessor_node->right);
+                predecessor_node = reinterpret_cast<TreeNode*>(tree->_one_dimensional_array->_data[predecessor]);
+            }
+
+            if (predecessor_node->right == Py_None) {
+                // Make the current node the right child of the predecessor
+                predecessor_node->right = node;
+                current = PyLong_AsLong(current_node->left);
+            } else {
+                // Revert the changes made to the tree
```

Comment:
Add a test to make sure that the tree as it is after performing Morris in order traversal.